src/Numerics/Mesh/Filters.jl

```diff
@@ -126,6 +126,37 @@ function spectral_filter_matrix(r, Nc, σ)
     V * Diagonal(Σ) / V
 end
 
+"""
+    modified_filter_matrix(r, Nc, σ)
+
+Returns the filter matrix that takes function values at the interpolation
+`N+1` points, `r`, converts them into Legendre polynomial basis coefficients,
+multiplies
+```math
+σ((n-N_c)/(N-N_c))
+```
+against coefficients `n=Nc:N` and evaluates the resulting polynomial at the
+points `r`. Unlike spectral_filter_matrix, this allows for the identity matrix,
+to be applied.
+"""
+function modified_filter_matrix(r, Nc, σ)
+    N = length(r) - 1
+    T = eltype(r)
+
+    @assert N >= 0
+    @assert 0 <= Nc
+
+    a, b = GaussQuadrature.legendre_coefs(T, N)
+    V = (N == 0 ? ones(T, 1, 1) : GaussQuadrature.orthonormal_poly(r, a, b))
+
+    Σ = ones(T, N + 1)
+    if Nc ≤ N
+        Σ[(Nc:N) .+ 1] .= σ.(((Nc:N) .- Nc) ./ (N - Nc))
+    end
+
+    V * Diagonal(Σ) / V
+end
```

Comment:
```suggestion
function modified_filter_matrix(r, Nc, σ)
    N = length(r) - 1
    T = eltype(r)

    @assert N >= 0
    @assert 0 <= Nc
    
    Nc > N && return Array{T}(I, N+1, N+1)

    a, b = GaussQuadrature.legendre_coefs(T, N)
    V = (N == 0 ? ones(T, 1, 1) : GaussQuadrature.orthonormal_poly(r, a, b))

    Σ = ones(T, N + 1)
    Σ[(Nc:N) .+ 1] .= σ.(((Nc:N) .- Nc) ./ (N - Nc))

    V * Diagonal(Σ) / V
end
```
Should we just force the identity matrix?

Reply:
I like this suggestion!